.github/workflows/test.yaml

```diff
@@ -105,3 +113,33 @@ jobs:
       run: |
         python --version
         tox -q
+
+    - name: upload dist
+      uses: actions/upload-artifact@v2
+      with:
+        name: py${{ matrix.python-version }}_dist
+        path: dist
+
+
+  all-successful:
+    # https://github.community/t/is-it-possible-to-require-all-github-actions-tasks-to-pass-without-enumerating-them/117957/4?u=graingert
+    runs-on: ubuntu-20.04
+    needs: [testing]
+    steps:
+      - name: Download dists for PyPI
+        uses: actions/download-artifact@v2
+        with:
+          name: py3.9_dist
```

Comment:
is this ok?

Reply:
Why was this marked as resolved? Just asking.

But I hope this can be implemented without upload + download of artifacts.

Reply:
ah I thought you were pointing out a bug - I was uploading artefacts in the wrong build stage